### ISSUE TITLE
Change request to make InvalidSheetNameException more descriptive

### ISSUE THREAD

**Issue by cmanley** [box/spout#362]
**Box\Spout\Writer\Common\Sheet.php**

**Background:**
I was setting sheet names to year month periods such as 201611 (an int) and getting InvalidSheetNameException exceptions. I traced it down to being the !== operator in doesContainInvalidCharacters() which may be something for a different issue report.

**Change request:**
I could create a fork and a pull request, but this is fairly short...

Replace first 2 lines of setName() with:
```php
if (!$this->isNameValid($name, $error_reason)) {
            $errorMessage = "The sheet's name is invalid ($error_reason). It did not meet at least one of these requirements:\n";
```
Note the $error_reason which receives the reason and prevents unnecessary hair pulling.

Replace isNameValid() with:
```php
    /**
     * Returns whether the given sheet's name is valid.
     * @see Sheet::setName for validity rules.
     *
     * @param string $name
     * @param string &$error_reason optional, receives reason on error.
     * @return bool TRUE if the name is valid, FALSE otherwise.
     */
    protected function isNameValid($name, &$error_reason = null)
    {
        if (!(is_string($name) || is_int($name))) {
        	$error_reason = 'not a string or int';
            return false;
        }
        $nameLength = $this->stringHelper->getStringLength($name);
        if (!$nameLength) {
        	$error_reason = 'zero length';
        	return false;
        }
        if ($nameLength > self::MAX_LENGTH_SHEET_NAME) {
        	$error_reason = "length $nameLength is too long";
        	return false;
        }
        if ($this->doesContainInvalidCharacters($name)) {
        	$error_reason = 'contains invalid characters';
        	return false;
        }
        if (!$this->isNameUnique($name)) {
        	$error_reason = 'not unique';
        	return false;
        }
        if ($this->doesStartOrEndWithSingleQuote($name)) {
        	$error_reason = 'starts or ends with single quote';
        	return false;
        }
        return true;
    }
```

**Comment by adrilo** [comment]
Hi @cmanley,

Thanks for your detailed suggestion. It seems like I took a shortcut when building the error messaging for this feature :) I'll use your proposed code to improve the messaging.

BTW if you passed an int, the error came from the `is_string()` check and not the  `!==` operator in `doesContainInvalidCharacters()` (that's at least what it should do and what it did when I tried with a sheet name: `(int)201611`.
